git-clone/entrypoint.sh

```diff
@@ -33,6 +33,13 @@ echo "https://oauth2:${GITLAB_OAUTH_TOKEN}@${GITLAB_HOST}" > .git/credentials
 git remote add origin $REPOSITORY
 git fetch origin
 
+git config http.proxy http://localhost:8080
+# Note: The proxy will still verify the certificates of
+#       the connection to the git server.
+git config http.sslVerify false
+git config --unset credential.helper
+rm .git/credentials
```

Comment:
At the moment the initial cloning doesn't go through the proxy since the init container has to complete before the other ones (including the sidecar proxy) are started, so didn't touch this logic for the moment and just made sure to clean up afterwards